### PR TITLE
Add few pal APIs to pal library

### PR DESCRIPTION
**Pull request by fieryorc**
Add few more APIs to the pal library. These are APIs needed for debugger code. We would like to contribute back and keep our changes minimal. 

@mikem8361, can you review this? Thanks.

### CODE REVIEW

**Comment by mikem8361** [comment]
LGTM

**Comment by mikem8361** [comment]
@jkotas or @janvorli do you guys have any problem with adding these APIs?

File: src/pal/src/misc/interlock.cpp [hunk 1]
```diff
@@ -269,6 +269,24 @@ InterlockedOr(
     return __sync_fetch_and_or(Destination, Value);
 }
 
+LONG
+PALAPI
```
Comment (jkotas) [inline src/pal/src/misc/interlock.cpp]: The Windows return type for these is BOOLEAN

**Comment by jkotas** [comment]
Also, it would be nice to add tests for these.

File: src/pal/src/misc/interlock.cpp (hunk 1 above)
Comment (janvorli) [inline src/pal/src/misc/interlock.cpp]: MSDN says the return type is UCHAR

**Comment by janvorli** [comment]
LGTM modulo @jkotas comments.

**Comment by fieryorc** [comment]
Can you take a quick look? I have updated the code with code review comments and added tests. I ran the tests manually, and they pass. @mikem8361 , @jkotas, @janvorli

**Comment by janvorli** [comment]
LGTM

**Comment by jkotas** [comment]
@fieryorc Could you please squash the commits before it is merged?

**Comment by jkotas** [comment]
LGTM otherwise

**Comment by mikem8361** [comment]
LGTM

**Comment by fieryorc** [comment]
Thanks. Merged (to single commit), Rebased on top, and commited.

**Pull request merged**